### PR TITLE
add missing cancel() for CallProfiler

### PR DESCRIPTION
**Pull request by tigercosmos**
add the missing `cancel()` for `CallProfiler`.

This PR should be merged before #315.

### CODE REVIEW

File: gtests/test_nopython_callprofiler.cpp [hunk 1]
```diff
@@ -191,5 +191,26 @@ TEST_F(CallProfilerTest, simple_case_2)
...
+    pProfiler->reset();
+
+    auto test1 = [&]()
+    {
+        USE_CALLPROFILER_PROFILE_THIS_FUNCTION();
+
+        auto test2 = [&]()
+        {
+            USE_CALLPROFILER_PROFILE_THIS_FUNCTION();
+            pProfiler->cancel();
```
Comment (tigercosmos) [inline gtests/test_nopython_callprofiler.cpp:205]: calling cancel here.

File: cpp/modmesh/toggle/RadixTree.hpp [hunk 2]
```diff
@@ -251,11 +251,19 @@ class CallProfiler
     /// Reset the profiler
     void reset();
 
+    /// Cancel
+    void cancel()
```
Comment (tigercosmos) [inline cpp/modmesh/toggle/RadixTree.hpp:255]: add the missing `cancel()`

**Comment by tigercosmos** [comment]
@yungyuc  please take a look.

File: cpp/modmesh/toggle/RadixTree.hpp [hunk 3]
```diff
@@ -227,6 +226,7 @@ class CallProfiler
     // Called when a function starts
     void start_caller(const std::string & caller_name, std::function<void()> cancel_callback)
     {
+        m_cancel_callbacks.push_back(cancel_callback);
```
Comment (tigercosmos) [inline cpp/modmesh/toggle/RadixTree.hpp:229]: putting the callback into the list.

File: cpp/modmesh/toggle/RadixTree.hpp [hunk 4]
```diff
@@ -251,11 +251,22 @@ class CallProfiler
     /// Reset the profiler
     void reset();
 
+    /// Cancel the profiling from all probes
+    void cancel()
+    {
+        for (auto & cancel_callback : m_cancel_callbacks)
+        {
+            cancel_callback();
```
Comment (tigercosmos) [inline cpp/modmesh/toggle/RadixTree.hpp:259]: cancel all probes.

**Review by yungyuc — APPROVED** [review]
LGTM

**Pull request merged**